### PR TITLE
[Dataset] Builtin GINDataset

### PR DESCRIPTION
**Pull request by hetong007**
## Description
Update the builtin GINDataset.
https://github.com/dmlc/dgl/issues/1776

## Checklist
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented
- [X] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [X] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/data/gindt.py [hunk 1]
```diff
@@ -198,11 +192,11 @@ def _load(self):
                 else:
                     nattrs = None
 
-                g.ndata['label'] = F.tensor(np.asarray(nlabels))
+                g.ndata['label'] = F.tensor(nlabels)
                 if len(self.nlabel_dict) > 1:
                     self.nlabels_flag = True
 
-                assert g.number_of_nodes() == n_nodes
+                assert len(g) == n_nodes
```
Comment (classicsong) [inline python/dgl/data/gindt.py:199]: It is better to use g.number_of_nodes()
len(g) will be deprecated soon.

File: python/dgl/data/gindt.py [hunk 2]
```diff
@@ -42,19 +34,26 @@ class GINDataset(object):
...
+    Returns
+    -------
+    GINDataset object as an iterable for dataloader.
+    Examples
+    --------
+    >>> data = GINDataset(name='MUTAG', self_loop=False)
+    >>> data[0]
+    (Graph(num_nodes=23, num_edges=54,
+           ndata_schemes={'label': Scheme(shape=(), dtype=torch.int64), 'attr': Scheme(shape=(7,), dtype=torch.float64)}
+           edata_schemes={}), 0)
```
Comment (classicsong) [inline python/dgl/data/gindt.py:46]: does the example be
for g in data:
    g... ?

File: python/dgl/data/gindt.py [hunk 3]
```diff
@@ -79,58 +78,53 @@ def __init__(self, name, self_loop, degree_as_nlabel=False):
...
-
         Paramters
         ---------
         idx : int
             The sample index.
-
         Returns
         -------
-        (dgl.DGLGraph, int)
+        (dgl.Graph, int)
```
Comment (classicsong) [inline python/dgl/data/gindt.py:108]: @jermainewang Is this DGLGraph or Graph?

Comment (hetong007) [reply python/dgl/data/gindt.py:108]:
```
>>> import dgl
Using backend: pytorch
>>> g = dgl.graph([])
>>> g
Graph(num_nodes=0, num_edges=0,
      ndata_schemes={}
      edata_schemes={})
```

File: python/dgl/data/gindt.py (hunk 2 above)
Comment (hetong007) [reply python/dgl/data/gindt.py:46]: Should we unify the example format in a later round of docstring update?

File: python/dgl/data/gindt.py (hunk 3 above)
Comment (classicsong) [reply python/dgl/data/gindt.py:130]: I check with mufei, this maybe some inconsistency with DGLGraph and DGLHeteroGraph. 
Let's left it there and fix it later.

**Pull request merged**